### ISSUE TITLE
Roadmap CRAN

### ISSUE THREAD

**Issue by lorenzwalthert** [curso-r/treesnip#23]
Do you have any plans to submit this to CRAN? If yes, when? Thanks :smile:

**Comment by dfalbel** [comment]
We don't have plans to submit to CRAN yet. Both catboost and lightgbm are not available on CRAN, and this blocks us for submitting treesnip.

**Comment by lorenzwalthert** [comment]
Right, lightgbm will make it to CRAN soon hopefully (https://github.com/microsoft/LightGBM/issues?q=CRAN), for catboost I don't think it's a priority for them at the moment unfortunately. At least judging from https://github.com/catboost/catboost/issues/439

**Comment by mdancho84** [comment]
Anyway you can submit without support for `catboost`? The `LightGBM` component is a big one.

**Comment by dfalbel** [comment]
Yes, the best way to do this IMHO is to move the lightgbm functionality out of treesnip into a new package and make treesnip depend on it. @Athospd what do you think?

**Comment by lorenzwalthert** [comment]
Or if everyone upvotes/comments here, they might prioritize it: https://github.com/catboost/catboost/issues/439

**Comment by Athospd** [comment]
If it is conveninent, I'm in! We can estipulate a due date for catboost give a sign toward CRAN release. CASE WHEN there is no updates from their team until november, 1th THEN we act. What about it?

What do you guys have in mind? Split this package in two: catboostnip and lightgbmnip?

**Comment by lorenzwalthert** [comment]
I don't think November is realistic. See how long it takes for lightgbm, and people are actively working on it. I'd wait until lightgbm is on CRAN then evaluate.

**Comment by jameslamb** [comment]
I found this issue via GitHub search. Wanted to let you all know that `{lightgbm}` is now on CRAN.

https://cran.r-project.org/web/packages/lightgbm/index.html

The package has been up for 10 days and we haven't received any emails from CRAN complaining of issues, so I think there is a good chance it will stay there this time. We are not announcing this widely in documentation or on social media until next week, but I think you folks here can start coding against the assumption that `{lightgbm}` is on CRAN 😀
